File: examples/python313-task-worker/src/task.py
from __future__ import annotations

import json
import os
import time
from typing import Any


def _to_dict(event: Any) -> dict[str, Any]:
    if isinstance(event, dict):
        return event
    if isinstance(event, str):
        try:
            parsed = json.loads(event)
            if isinstance(parsed, dict):
                return parsed
        except Exception:
            return {"raw": event}
    return {}


def _clamp_sleep_ms(value: Any) -> int:
    try:
        parsed = int(value)
    except Exception:
        return 0
    if parsed <= 0:
        return 0
    return min(parsed, 120000)


def handler(event: Any, context: Any):
    payload = _to_dict(event)
    job = payload.get("job") if isinstance(payload.get("job"), str) and str(payload.get("job")).strip() else "demo-job"
    mode = payload.get("mode") if isinstance(payload.get("mode"), str) else "ok"
    attempt_value = payload.get("attempt", 1)
    try:
        attempt = int(attempt_value)
    except Exception:
        attempt = 1
    sleep_ms = _clamp_sleep_ms(payload.get("sleepMs", 0))
    metadata = payload.get("metadata") if isinstance(payload.get("metadata"), dict) else {}

    if mode == "sleep" and sleep_ms > 0:
        time.sleep(sleep_ms / 1000.0)

    return {
        "ok": True,
        "service": "python313-task-worker",
        "runtime": os.getenv("LICELL_FC_RUNTIME") or os.getenv("FC_RUNTIME") or os.getenv("RUNTIME") or "python3.13",
        "job": job,
        "mode": mode,
        "attempt": attempt,
        "sleepMs": sleep_ms,
        "metadata": metadata,
        "now": time.strftime("%Y-%m-%dT%H:%M:%SZ", time.gmtime()),
    }

File: examples/node22-task-worker/src/task.ts
function toRecord(input: unknown): Record<string, unknown> {
  if (input && typeof input === 'object' && !Array.isArray(input)) {
    return input as Record<string, unknown>;
  }
  if (typeof input === 'string') {
    try {
      const parsed = JSON.parse(input);
      if (parsed && typeof parsed === 'object' && !Array.isArray(parsed)) {
        return parsed as Record<string, unknown>;
      }
    } catch {
      return { raw: input };
    }
  }
  return {};
}

function sleep(ms: number) {
  return new Promise((resolve) => setTimeout(resolve, ms));
}

function clampSleepMs(input: unknown) {
  const value = Number(input ?? 0);
  if (!Number.isFinite(value) || value <= 0) return 0;
  return Math.min(Math.floor(value), 120000);
}

export async function handler(event: unknown) {
  const payload = toRecord(event);
  const job = typeof payload.job === 'string' && payload.job.trim() ? payload.job.trim() : 'demo-job';
  const mode = typeof payload.mode === 'string' ? payload.mode : 'ok';
  const sleepMs = clampSleepMs(payload.sleepMs);
  const attempt = Number.isFinite(Number(payload.attempt)) ? Number(payload.attempt) : 1;
  const metadata = payload.metadata && typeof payload.metadata === 'object' && !Array.isArray(payload.metadata)
    ? payload.metadata
    : {};

  if (mode === 'sleep' && sleepMs > 0) {
    await sleep(sleepMs);
  }

  return {
    ok: true,
    service: 'node22-task-worker',
    runtime: process.env.LICELL_FC_RUNTIME || process.env.FC_RUNTIME || process.env.RUNTIME || 'nodejs22',
    job,
    mode,
    attempt,
    sleepMs,
    metadata,
    now: new Date().toISOString()
  };
}
